From 0f96a197b69410f357671c63da4648432c5df939 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:06 +0200
Subject: [PATCH 21/62] VideoPlayer: fix race condition in Get(StreamType)

After Count() has been called, streams may be added or removed,
leading to bogus results, because the mutex is released during the
loop.

Fix this by adding another recursive lock.

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 xbmc/cores/VideoPlayer/VideoPlayer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/cores/VideoPlayer/VideoPlayer.cpp b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
index f909c8e..fa05920 100644
--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
@@ -120,6 +120,7 @@ SelectionStream& CSelectionStreams::Get(StreamType type, int index)
 
 std::vector<SelectionStream> CSelectionStreams::Get(StreamType type)
 {
+  CSingleLock lock(m_section);
   std::vector<SelectionStream> streams;
   std::copy_if(m_Streams.begin(), m_Streams.end(), std::back_inserter(streams),
     [type](const SelectionStream &stream)
-- 
2.7.4

